packages/core/src/provider/base.ts

```diff
@@ -470,13 +473,18 @@ export class SmartAccountProvider<
       );
     }
 
-    request.signature = (await this.account.signUserOperationHash(
-      getUserOperationHash(
-        request,
-        this.getEntryPointAddress(),
-        BigInt(this.chain.id)
-      )
-    )) as `0x${string}`;
+    const userOpHash = getUserOperationHash(
+      request,
+      this.getEntryPointAddress(),
+      BigInt(this.chain.id)
+    );
+
+    request.signature =
+      overrides?.signer != null
```

Comment:
oh wait... what's goin on here? why do we need to override the signer ever?

this should be done by just connecting the provider to a new instance of an account